src/debug/FuncDebugProviderBase.ts

```diff
@@ -46,7 +46,7 @@ export abstract class FuncDebugProviderBase implements DebugConfigurationProvide
             context.telemetry.suppressIfSuccessful = true;
 
             this._debugPorts.set(folder, <number | undefined>debugConfiguration.port);
-            if (debugConfiguration.preLaunchTask === hostStartTaskName) {
+            if (hostStartTaskNameRegExp.test(debugConfiguration.preLaunchTask)) {
```

Comment:
I'm okay with this change, but I don't think people should have been changing the name of their `preLaunchTask` to include the port. Where did you see that this was an issue? I didn't notice it skimming through the original issue.

Reply:
User mentioned: 

`The issue is only with the local.settings.json, it only seems to look for/pick the local.settings.json file of the first function app in the list.`

When I started digging into the screenshot and testing out the different configs, I stumbled across this as the likely culprit.